access_request.go

```diff
@@ -41,3 +44,7 @@ func NewAccessRequest(session Session) *AccessRequest {
 func (a *AccessRequest) GetGrantTypes() Arguments {
 	return a.GrantTypes
 }
+
+func (a *AccessRequest) SetDelegatingClient(client Client) {
```

Comment:
I guess you should move this to an interface as well? `TokenExchangeAccessRequestor`?

Reply:
Do you mean by extending the AccessRequest interface?
What would be the advantage of that?

Reply:
This is impersonation only.

Reply:
Ok, so this would set the OAuth2 Client which the `subject_token` was granted for in the Token Exchange request, correct? If so, maybe we rename it to `SetSubjectTokenClient` to allow further extension (in the future) for `SetActorTokenClient`? And then I think it makes sense to define another interface as @mitar suggested:


```
// in handler/rfc8693/access_request.go

type interface {
  SetDelegatingClient(client Client)
}
```

Reply:
Extracted interface but kept it in same file/package due to otherwise resulting cyclical dependencies.